src/repo.rs

```diff
@@ -59,6 +59,24 @@ impl Repository {
         }
     }
 
+    /// Find and open a repository, with additional options.
+    pub fn open_ext<P: AsRef<Path>>(path: P,
+                                    flags: RepositoryOpenFlags,
+                                    ceiling_dirs: Option<&str>)
```

Comment:
I believe that this is actually a naturally-separated list of directories, so perhaps this could be an iterator of `T: AsRef<OsStr>`? I think there's some other APIs for consuming an iterator like that elsewhere as well.